app/src/main/java/com/gazlaws/codeboard/CodeBoardIME.java

```diff
@@ -342,6 +355,19 @@ public void onViewClicked(boolean focusChanged) {
     }
 
     public void onRelease(int primaryCode) {
+
+        /*
+         *   After the space button is released,
+         *   we check whether it was long pressed or not.
+         *   If it was, we don't do anything,
+         *   but If it wasn't, we print a "space" to the screen.
+         */
+        if ((primaryCode == 32) && (! longPressedSpaceButton)) {
+
+            InputConnection ic = getCurrentInputConnection ();
+            ic.commitText (String.valueOf ((char) primaryCode), 1);
```

Comment:
Can you move the longPressedSpaceButton = false; from line 190 to line 369? It's better to keep the logic close together.

Reply:
Did it!